react/week4/my-app/src/Inputuser.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect, useCallback } from "react";
+
+import ShowUser from "./showusers";
+
+const InputUser = () => {
+  const [name, setName] = useState("");
+
+  const [users, setUsers] = useState([]);
+
+  const [isLoading, setIsLoading] = useState(true);
+  
+  const [isEmpty, setEmpty] = useState(true);
```

Comment:
To minimize the amount of states I would recommend checking to see if user.length > 0 further down instead of a separate state.